rest_framework_api_key/models.py

```diff
@@ -21,12 +20,12 @@ def create_key(self, **kwargs) -> Tuple["APIKey", str]:
 
     def is_valid(self, key: str) -> bool:
         prefix, _, _ = key.partition(".")
-
         try:
             obj = self.get(id__startswith=prefix, revoked=False)
         except self.model.DoesNotExist:
             return False
-
+        if obj.has_expired:
```

Comment:
Sorry if I was not clear in my previous comment — I think we should compute has_expired based on the expiry_date set on the APIkey object. It shouldn’t be another field, since and the revoked field already serves the purposes of permanently deactivating the key.